src/interactions/panZoomInteraction.ts

```diff
@@ -439,6 +577,117 @@ namespace Plottable.Interactions {
+    public minDomainValue<D>(quantitativeScale: QuantitativeScale<D>, minDomainValue?: D): any {
+      if (minDomainValue == null) {
+        return this._minDomainValues.get(quantitativeScale);
+      }
+      let maxValueForScale = this.maxDomainValue(quantitativeScale);
+      if (maxValueForScale != null && maxValueForScale.valueOf() < minDomainValue.valueOf()) {
+        throw new Error("minDomainValue must be smaller than maxDomainValue for the same Scale");
+      }
+      this._minDomainValues.set(quantitativeScale, minDomainValue);
+      return this;
+    }
+
+    /**
+     * Gets the maximum domain value for the scale, constraining the pan/zoom
+     * interaction to a maximum value in the domain.
+     *
+     * Note that this differs from minDomainExtent/maxDomainExtent, in that
+     * those methods provide constraints such as showing at least 2 but no more
+     * than 5 values at a time.
+     *
+     * By contrast, minDomainValue/maxDomainValue set a boundary beyond which
+     * the user cannot pan/zoom.
+     *
+     * @param {QuantitativeScale<any>} quantitativeScale The scale to query
+     * @returns {D} The maximum domain extent for the scale.
+     */
+    public maxDomainValue<D>(quantitativeScale: QuantitativeScale<D>): D;
+    /**
+     * Sets the maximum domain value for the scale, constraining the pan/zoom
+     * interaction to a maximum value in the domain.
+     *
+     * Note that this differs from minDomainExtent/maxDomainExtent, in that
+     * those methods provide constraints such as showing at least 2 but no more
+     * than 5 values at a time.
+     *
+     * By contrast, minDomainValue/maxDomainValue set a boundary beyond which
+     * the user cannot pan/zoom.
+     *
+     * @param {QuantitativeScale<any>} quantitativeScale The scale to query
+     * @param {D} maxDomainExtent The maximum domain extent for the scale.
+     * @returns {Interactions.PanZoom} The calling PanZoom Interaction.
+     */
+    public maxDomainValue<D>(quantitativeScale: QuantitativeScale<D>, maxDomainValue: D): this;
+    public maxDomainValue<D>(quantitativeScale: QuantitativeScale<D>, maxDomainValue?: D): any {
+      if (maxDomainValue == null) {
+        return this._maxDomainValues.get(quantitativeScale);
+      }
+      let minValueForScale = this.minDomainValue(quantitativeScale);
+      if (minValueForScale != null && maxDomainValue.valueOf() < minValueForScale.valueOf()) {
+        throw new Error("maxDomainValue must be larger than minDomainValue for the same Scale");
+      }
+      this._maxDomainValues.set(quantitativeScale, maxDomainValue);
+      return this;
+    }
+
+    /**
+     * Uses the current domain of each scale to apply a minimum and maximum
+     * domain value for that scale.
+     *
+     * This constrains the pan/zoom interaction to show no more than the domain
+     * of the scale.
+     */
+    public constrainToScaleDomainValues() {
```

Comment:
I didn't know what this method did until I read your comment. Can we just implement this on the blueprint-chart side?